### ISSUE TITLE
Request of Implementation of lgamma

### ISSUE THREAD

**Issue by c0nleyinnnn** [ml-explore/mlx#2030]
I am trying to port an mlx implementation of the SCVI module of scvi-tools (a comprehensive de-batching approach using VAE models), which currently overcomes the problem of mlx.array not supporting scipy sparse matrices by converting the sparse matrices to dense matrices, but in practice I can only use python's internal lanczos approximation due to the lack of a native C++ backend lgamma function (something like torch.lgamma or jax.scipy.special.gamma) which leads to a severely limited code with high training loss, and I would like to request the implementation of a built in gamma as well as the lgamma function since I am completely lacking in C code capabilities.
